src/components/wallet/views/balanceDisplay.tsx

```diff
@@ -3,16 +3,22 @@ import { has as loHas } from 'lodash-es'
 import WalletButton from './walletButton'
 import { Wallet } from '../wallet'
 
-interface Balance {
-  balance: string
-  is_authorized: boolean
-  asset_type: string
-  asset_code: string
-  asset_issuer: string
-}
-
 export default function balanceDisplay(this: Wallet) {
-  const balanceRow = (balance: Balance) => {
+  let totalBalances = []
+  // Combines the two arrays of different types
+  // to allow balanceRow to iterate and map
+  // their values in the view
+  if (this.account.state) {
```

Comment:
If this view is being rendered, won't `this.account.state` always be populated?

Reply:
Funny enough no.
When you first create an account state isnt populated yet.
Heres the call stack (sumarized a lot)
```
balanceDisplay (stellar-wallet.entry.js:49759)
loggedInContent (stellar-wallet.entry.js:49849)
render (stellar-wallet.entry.js:49891)
callRender (index-9c8102b3.js:1415)
updateComponent (index-9c8102b3.js:1348)
createAccount (stellar-wallet.entry.js:49901)
onClick (stellar-wallet.entry.js:49872)
```

In create account we dont populate the state
```
this.account = {
      publicKey: keypair.publicKey(),
      secretKey: keypair.secret(),
    };
    this.assets.set('XLM', {});
    set1('WALLET[keystore]', btoa(JSON.stringify(this.account)));
    ```
    So when loggedin content calls `balanceDisplay.call(this)`
    it will panic if we dont make this check.
    Before we were able to avoid this check due to just calling the state directly here 
    ```
      return (
    <div>
      {loHas(this.account, 'state') ? (
        <pre class="account-state">
          <h2 class="balance-headers">Balances</h2>
          {(this.account.state.balances as Balance[]).map(balanceRow)}
        </pre>
      ) : null}
    </div>
  )
  ```
  Where if there is no state we simply return null.
  However since we're building a concatenated list from both trusted assets and untrusted before we reach the return call that `{loHas(this.account, 'state')` ternary op will not really matter (I guess we could refactor it out now that I think about it).